addons/website/tests/test_page.py

```diff
@@ -303,3 +303,21 @@ def test_homepage_not_slash_url(self):
         root_html = html.fromstring(r.content)
         canonical_url = root_html.xpath('//link[@rel="canonical"]')[0].attrib['href']
         self.assertEqual(canonical_url, website.domain + "/")
+
+    def test_t_cache_footer_debug_link(self):
+        # Debug link (flag) should be use the current url.
+        self.url_open('/')
+        self.url_open('/contactus')
+        r = self.url_open('/?debug=1')
+        self.assertIn('<a class="o_debug_mode" href="?debug="', r.text)
+        r = self.url_open('/contactus?debug=1')
+        self.assertIn('<a class="o_debug_mode" href="?debug="', r.text)
```

Comment:
nitpicking: We could use the second url_open to test that the query are correctly preserved as well
```suggestion
        self.url_open('/')
        self.url_open('/contactus')
        r = self.url_open('/?debug=1')
        self.assertIn('<a class="o_debug_mode" href="?debug="', r.text)
        r = self.url_open('/contactus?name=MyName')  # don't force debug here, already done on previous step
        self.assertIn('<a class="o_debug_mode" href="?debug=&name=MyName"', r.text)
```